nemo/collections/nlp/models/language_modeling/megatron_lm_encoder_decoder_model.py

```diff
@@ -405,20 +360,36 @@ def training_step(self, batch, batch_idx):
             if loss_scale is not None:
                 self.log('loss_scale', loss_scale)
 
-        self.log('reduced_train_loss', loss_mean, prog_bar=True, rank_zero_only=True)
+        self.log(
+            'reduced_train_loss', loss_mean, prog_bar=True, rank_zero_only=True, batch_size=self.cfg.global_batch_size
+        )
         lr = self._optimizer.param_groups[0]['lr']
-        self.log('lr', lr, rank_zero_only=True)
-        self.log('global_step', self.trainer.global_step, prog_bar=True, rank_zero_only=True)
+        self.log('lr', lr, rank_zero_only=True, batch_size=self.cfg.global_batch_size)
+        self.log(
+            'global_step',
+            self.trainer.global_step,
+            prog_bar=True,
+            rank_zero_only=True,
+            batch_size=self.cfg.global_batch_size,
```

Comment:
should we set batch_size=1? I am not sure how it affects things on PTL side

Reply:
Followed up with @aklife97 I need to investigate what this really does.

@awaelchli @carmocca when using `dataloader_iter` and setting the `batch_size` in `self.log`, what does this batch size do? Is it ok to set to 1?

Reply:
Hey @SeanNaren 

I think you are setting this here because for the special case of passing dataloader_iter to training_step, there is a warning in Lightning saying: 

> With `def training_step(self, dataloader_iter)`, `self.log(..., batch_size=...)` should be provided

You can set it to 1 to avoid the warning. Normally Lightning uses the batch size to weight each log when averaging it (on_epoch=True). If you know that all batch sizes are the same, or you don't care about the averaging, then the value can be arbitrary here (e.g. 1).

I'm not 100% sure why it is required for the dataloader_iter case, but I think it is just that Lightning won't be able to infer the batch size automatically otherwise.

Reply:
Thanks @awaelchli, @SeanNaren 

@ericharper, FYI

Reply:
> I'm not 100% sure why it is required for the dataloader_iter case

This is because lightning needs access to the batch to get its size.
But with `dataloader_iter`, the dataloader is consumed in the `training_step` directly, so lightning doesn't have a reference to the batch.